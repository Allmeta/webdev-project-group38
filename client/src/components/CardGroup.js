import React, { Component } from 'react'
import MovieCard from './Card.js'
import { fetchMovies } from '../actions/MovieActions'
import { connect } from 'react-redux'
import styled from 'styled-components'

const CardWrapper = styled.div`
  display:inline-grid;
  grid-template-columns: repeat(3,1fr);
  grid-column-gap: 30px;
  grid-row-gap: 30px;
  margin: 0 auto;
  max-width: 80vw;
  
  @media (max-width: 1330px){
    grid-template-columns: repeat(2,1fr);
  }
  @media (max-width: 860px){
    grid-template-columns: repeat(1,1fr);
    max-width: 100vw;
    grid-row-gap: 20px;
  }
`

class CardGroup extends Component {
  constructor (props) {
    super(props)

    this.state = {
      items: props.items // provided by connect@mapStateToProps
    }
  }

  componentDidMount () {
    // const { fetchMovies } = this.props
    // fetchMovies()
  }

  render () {
    const image_base = 'https://image.tmdb.org/t/p/w500'

    const items = this.props.items

    return (
      <CardWrapper>
        {items && items.map((o) =>
          <MovieCard
            key={o.movie_id}
            img={image_base + o.poster_path}
            description={o.summary}
            title={o.title}
            popularity={Number(o.rating)}
            comment={o.comment}
            date={o.release_date}
            language={o.language}
            // genres={o.genres.join(' ')}
            genres={['WIP1', 'WIP2', 'WIP3']}
          />
        )}
      </CardWrapper>
    )
  }
}

function mapStateToProps (state) {
  return {
    items: state.MovieReducer.items
  }
}

function mapDispatchToProps (dispatch) {
  return {
    fetchMovies: (data) => dispatch(fetchMovies(data))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(CardGroup)
