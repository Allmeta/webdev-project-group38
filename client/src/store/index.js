import { createStore, applyMiddleware, compose } from 'redux'
import { SearchFormReducer } from "../reducers/SearchFormReducer";
import thunk from 'redux-thunk'

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(SearchFormReducer, composeEnhancers(
  applyMiddleware(thunk)
));

export default store;