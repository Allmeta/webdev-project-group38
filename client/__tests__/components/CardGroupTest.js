import React from 'react'
import CardGroup from '../../src/components/CardGroup'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { Provider } from 'react-redux'
import store from '../../src/store/index.js'

it('renders correctly', () => {
  const tree = renderer.create(<Provider store={store}><CardGroup></CardGroup></Provider>).toJSON()
  expect(tree).toMatchSnapshot()
})
