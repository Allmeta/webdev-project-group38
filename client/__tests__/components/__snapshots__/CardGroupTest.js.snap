// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  display: inline-grid;
  grid-template-columns: repeat(3,1fr);
  grid-column-gap: 30px;
  grid-row-gap: 30px;
  margin: 0 auto;
  max-width: 80vw;
}

@media (max-width:1330px) {
  .c0 {
    grid-template-columns: repeat(2,1fr);
  }
}

@media (max-width:860px) {
  .c0 {
    grid-template-columns: repeat(1,1fr);
    max-width: 100vw;
    grid-row-gap: 20px;
  }
}

<div
  className="c0"
/>
`;
